staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -304,6 +297,109 @@ func (w *WebhookAuthorizer) Authorize(ctx context.Context, attr authorizer.Attri
 
 }
 
+func resourceAttributesFrom(attr authorizer.Attributes) *authorizationv1.ResourceAttributes {
+	ret := &authorizationv1.ResourceAttributes{
+		Namespace:   attr.GetNamespace(),
+		Verb:        attr.GetVerb(),
+		Group:       attr.GetAPIGroup(),
+		Version:     attr.GetAPIVersion(),
+		Resource:    attr.GetResource(),
+		Subresource: attr.GetSubresource(),
+		Name:        attr.GetName(),
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AuthorizeWithSelectors) {
+		// If there's an error while parsing selectors, we can safely ignore the selector
+		// This is because selectors only narrow, so if broader permission is available, the request can be allowed.
+		if selectorRequirements, err := fieldSelectorToAuthorizationAPI(attr); err == nil && selectorRequirements != nil {
+			ret.FieldSelector = &authorizationv1.FieldSelectorAttributes{
+				Requirements: selectorRequirements,
+			}
+		}
+
+		if selectorRequirements, err := labelSelectorToAuthorizationAPI(attr); err == nil && selectorRequirements != nil {
+			ret.LabelSelector = &authorizationv1.LabelSelectorAttributes{
+				Requirements: selectorRequirements,
+			}
+		}
+	}
+
+	return ret
+}
+
+func fieldSelectorToAuthorizationAPI(attr authorizer.Attributes) ([]authorizationv1.FieldSelectorRequirement, error) {
+	requirements, err := attr.ParseFieldSelector()
+	if err != nil {
+		return nil, err
+	}
+	if len(requirements) == 0 {
+		return nil, nil
+	}
+
+	retRequirements := []authorizationv1.FieldSelectorRequirement{}
+	for _, requirement := range requirements {
+		retRequirement := authorizationv1.FieldSelectorRequirement{}
+		switch {
+		case requirement.Operator == selection.Equals || requirement.Operator == selection.DoubleEquals || requirement.Operator == selection.In:
+			retRequirement.Operator = metav1.LabelSelectorOpIn
+			retRequirement.Key = requirement.Field
+			retRequirement.Values = []string{requirement.Value}
+		case requirement.Operator == selection.NotEquals || requirement.Operator == selection.NotIn:
+			retRequirement.Operator = metav1.LabelSelectorOpNotIn
+			retRequirement.Key = requirement.Field
+			retRequirement.Values = []string{requirement.Value}
+		default:
+			// TODO decide about failing versus ignoring. The ignoring is safe, but not optimal
+			continue
+		}
+		retRequirements = append(retRequirements, retRequirement)
+	}
+
+	if len(retRequirements) == 0 {
+		return nil, nil
```

Comment:
comment here that this means all requirements were for unknown operators and were dropped (same below)